### PR TITLE
Enhance filter logic

### PR DESCRIPTION
**Pull request by maxmoehl**
See individual commits.

### CODE REVIEW

File: src/pcap/pcap.go [hunk 1]
```diff
@@ -191,3 +193,52 @@ func vcapIDFromCtx(ctx context.Context) (*string, error) {
...
+
+// patchFilter extends the given filter by excluding the filter generated
+// by generateApiFilter.
+func patchFilter(filter string) (string, error) {
+	apiFilter, err := generateApiFilter()
+	if err != nil {
+		return "", err
+	}
+
+	if filter == "" {
```
Comment (peanball) [inline src/pcap/pcap.go:222]: Removing whitespace would be pretty good here, just in case.

File: src/pcap/pcap.go [hunk 2]
```diff
@@ -191,3 +193,52 @@ func vcapIDFromCtx(ctx context.Context) (*string, error) {
...
+		return fmt.Sprintf("not (%s)", apiFilter), nil
+	} else {
+		return fmt.Sprintf("not (%s) and (%s)", apiFilter, filter), nil
```
Comment (peanball) [inline src/pcap/pcap.go:225]: You could technically do the same thing with the `join` here as you do below.

File: src/pcap/pcap.go [hunk 3]
```diff
@@ -191,3 +193,65 @@ func vcapIDFromCtx(ctx context.Context) (*string, error) {
...
+	}
+	if len(addrs) == 0 {
+		return "", fmt.Errorf("unable to determine ip addresses")
+	}
+
+	var ips []string
+	for _, addr := range addrs {
+		ipNet, ok := addr.(*net.IPNet)
+		// check that:
+		// * ipNet is actually an IP address
```
Comment (peanball) [inline src/pcap/pcap.go:243]: should this not be `ip host 192.168.1.1 or 192.168.1.2`, etc.?
```suggestion
	return strings.Join(ipFilters, " or "), nil
```

**Review by peanball — COMMENTED** [review]
please check the join condition

File: src/pcap/pcap.go (hunk 2 above)
Comment (maxmoehl) [reply src/pcap/pcap.go:225]: No.

File: src/pcap/pcap.go [hunk 4]
```diff
@@ -191,3 +193,67 @@ func vcapIDFromCtx(ctx context.Context) (*string, error) {
...
+			ips = append(ips, ipNet.IP.String())
+		}
+	}
+
+	var ipFilters []string
+	for _, ip := range ips {
+		ipFilters = append(ipFilters, fmt.Sprintf("ip host %s", ip))
+	}
+
+	return strings.Join(ipFilters, " or "), nil
```
Comment (peanball) [inline src/pcap/pcap.go:258]: should this not be `ip host 192.168.1.1 or 192.168.1.2`, etc.?
```suggestion
	return strings.Join(ipFilters, " or "), nil
```

File: src/pcap/pcap.go (hunk 1 above)
Comment (peanball) [reply src/pcap/pcap.go:222]: thx

**Review by peanball — APPROVED** [review]

**Pull request merged**